test/regressions/http-compression.test.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @jest-environment node
+ * @see https://github.com/mswjs/interceptors/issues/109
+ */
+import got from 'got';
+import compression from 'compression';
+import * as zlib from 'zlib';
+
+import { createServer, ServerApi } from '@open-draft/test-server'
+import { createInterceptor, IsomorphicResponse } from '../../src'
+import { interceptClientRequest } from '../../src/interceptors/ClientRequest'
+
+let server: ServerApi
+
+const interceptor = createInterceptor({
+  modules: [interceptClientRequest],
+  resolver(req) {
+  },
+})
+
+beforeAll(async () => {
+  server = await createServer((app) => {
+    app.use(compression( { threshold: 1, }));
+    app.get('/', (req, res) =>
+      res.send({hello: 'world'}));
+  })
+
+  interceptor.apply()
+})
+
+afterAll(async () => {
+  interceptor.restore()
+  await server.close()
+})
+
+const GZIP_MAGIC_HEX = '1f8b08';
+
+test('returns a valid body', async () => {
+  const intercepted = new Promise<IsomorphicResponse>(resolve =>
+    interceptor.on('response', (req, resp) =>
+      resolve(resp)));
+  expect(await got(server.http.makeUrl('/')).json()).toEqual({hello: 'world'});
```

Comment:
I like how this test showcases real usage, but I find it having too many moving parts to be straightforward. I think we should get rid of as many dependencies as we can and narrow it down to the barebone case. For instance:

1. Replace `got` with `http`. 
2. Not sure why `zlib` is here. Looks like it handles the compressed buffer, but doesn't that introduce a middle-man in the logic? Unless the issue is zlib-specific.

Reply:
`zlib` is here to validate the returned response is valid, and not otherwise corrupted by the middleware. Unfortunately zlib throws an absolutely useless error ("invalid!") when it sees invalid data.

I don't plan to further contribute to this issue at this time, as the library has other bugs which mean that it is not useful to me, and which I can't easily reproduce outside of our application. e.g. it causes supertest to "hang" (never resolve) during some error situations.

Reply:
Thank you for your effort anyway, @FauxFaux. I'll see if this can be turned into a properly isolated test suite.